### ISSUE TITLE
Open without browser

### ISSUE THREAD

**Issue by birnstiel** [caleb531/open-conference-url#10]
Feature request:

There is a way to open the urls directly in zoom. As I mostly use zoom, I duplicated your `conf` part of the workflow and modified it like this:

<img width="729" alt="screenshot" src="https://user-images.githubusercontent.com/3997171/161007771-01d85b96-fabf-406b-9ccf-2a4deb25f059.png">

And the content of the `Run Script` part is this:

```bash
url="${event_conference_url/https/zoommtg}"
url="${url/\/j\///join?action=join&confno=}"
url="${url/?pwd=/&pwd=}";
open $url;
//echo $url;
echo $event_conference_url;
```

It converts the URL to one that can be directly opened in the zoom app.

**Comment by caleb531** [comment]
@birnstiel This is a fantastic idea! Your solution would need to be re-implemented slightly, since it would break non-Zoom URLs that need to remain `https`. However, I have an idea for how this could be done. I have added it to my todo list, and will reply back when I have something working.

**Comment by birnstiel** [comment]
Sorry, I wasn't clear, I meant to have this as an additional keyword. So `conf` works for everything, `zoom` works only for zoom meetings. Could change the filter to only show zoom meetings then, but I don't quite get how this filter works.

<img width="790" alt="Bildschirmfoto 2022-03-31 um 19 44 14" src="https://user-images.githubusercontent.com/3997171/161117484-c42db2d1-d11f-4a78-a35a-117dbe949bc5.png">

**Comment by caleb531** [comment]
@birnstiel I don't think a second keyword is needed. Rather, I can automatically convert all `https` Zoom URLs to the proper `zoommtg` protocols under the same `conf` keyword. All other (non-Zoom) conference URLs would be unaffected.

**Comment by birnstiel** [comment]
Yes, that works as well!

**Comment by caleb531** [comment]
@birnstiel I just released v4.0.0-beta.1 with some new logic to automatically convert Zoom URLs (per your provided algorithm) so as to bypass the browser. It can be disabled with the new `use_direct_zoom` preference (now as a workflow variable).

Can you please test it out and let me know if it's as you expect?

https://github.com/caleb531/open-conference-url/releases/tag/v4.0.0-beta.1

Thanks so much,
Caleb

**Comment by birnstiel** [comment]
Thanks for implementing this! Yes, it works for normal zoom links, but `my-company.zoom.us` still opens in the browser.

**Comment by caleb531** [comment]
@birnstiel Ah, my mistake! There was an additional regex I needed to modify. This is now fixed in v4.0.0-beta.2:

https://github.com/caleb531/open-conference-url/releases/tag/v4.0.0-beta.2

**Comment by caleb531** [comment]
@birnstiel Closing this issue since I just released v4.0.0 stable with the functionality and fixes we discussed here. Thanks so much for providing feedback on all this!

https://github.com/caleb531/open-conference-url/releases/tag/v4.0.0
